dev/RadialGradientBrush/RadialGradientBrush.cpp

```diff
@@ -0,0 +1,228 @@
+    else
+    {
+        // If CompositionRadialGradientBrush won't be used then listen for changes to the fallback color so the fallback brush can be updated.
+        m_fallbackColorChangedToken.value = RegisterPropertyChangedCallback(winrt::XamlCompositionBrushBase::FallbackColorProperty(), { this, &RadialGradientBrush::OnFallbackColorChanged });
+    }
+}
+
+void RadialGradientBrush::OnDisconnected()
+{
+    m_isConnected = false;
+
+    if (m_brush)
+    {
+        m_brush.Close();
+        m_brush = nullptr;
+        CompositionBrush(nullptr);
+    }
+
+    if (m_gradientStopsChangedToken)
+    {
+        m_gradientStops.VectorChanged(m_gradientStopsChangedToken);
+    }
+
+    if (m_fallbackColorChangedToken)
+    {
+        UnregisterPropertyChangedCallback(winrt::XamlCompositionBrushBase::FallbackColorProperty(), m_fallbackColorChangedToken.value);
+        m_fallbackColorChangedToken.value = 0;
+    }
+}
+
+void RadialGradientBrush::OnPropertyChanged(const winrt::DependencyPropertyChangedEventArgs& args)
+{
+    if (SharedHelpers::IsCompositionRadialGradientBrushAvailable())
+    {
+        auto property = args.Property();
+
+        if (property == s_EllipseCenterProperty)
+        {
+            UpdateCompositionGradientEllipseCenter();
+        }
+        else if (property == s_EllipseRadiusProperty)
+        {
+            UpdateCompositionGradientEllipseRadius();
+        }
+        else if (property == s_GradientOriginOffsetProperty)
+        {
+            UpdateCompositionGradientOriginOffset();
+        }
+        else if (property == s_MappingModeProperty)
+        {
+            UpdateCompositionGradientMappingMode();
+        }
+        else if (property == s_InterpolationSpaceProperty)
+        {
+            UpdateCompositionInterpolationSpace();
+        }
+    }
+}
+
+void RadialGradientBrush::OnFallbackColorChanged(const winrt::DependencyObject& sender, const winrt::DependencyProperty& args)
+{
+    UpdateFallbackBrush();
+}
+
+void RadialGradientBrush::EnsureCompositionBrush()
+{
+    if (m_isConnected && !m_brush)
+    {
+        auto compositor = winrt::Window::Current().Compositor();
+
+        if (SharedHelpers::IsCompositionRadialGradientBrushAvailable())
+        {
+            // If CompositionRadialGradientBrush is available then use it to render a gradient.
+            m_brush = compositor.CreateRadialGradientBrush();
+
+            UpdateCompositionGradientEllipseCenter();
+            UpdateCompositionGradientEllipseRadius();
+            UpdateCompositionGradientOriginOffset();
+            UpdateCompositionGradientStops();
+            UpdateCompositionGradientMappingMode();
+            UpdateCompositionInterpolationSpace();
+        }
+        else
+        {
+            // If CompositionRadialGradientBrush isn't available then render using the FallbackColor.
+            m_brush = compositor.CreateColorBrush();
+            UpdateFallbackBrush();
+        }
+
+        CompositionBrush(m_brush);
```

Comment:
>CompositionBrush [](start = 8, length = 16)

what if someone has set this DP? Is that just not supported?

Reply:
CompositionBrush is just a simple property and is protected

Reply:
oh, cool.